consensus/state_sync/module.go

```diff
@@ -75,3 +193,47 @@ func (m *stateSync) GetBus() modules.Bus {
 func (m *stateSync) GetModuleName() string {
 	return stateSyncModuleName
 }
+
+// metadataSyncLoop periodically sends metadata requests to its peers to collect &
+// aggregate metadata related to synching the state.
+// It is intended to be run as a background process via a goroutine.
+func (m *stateSync) metadataSyncLoop() error {
+	logger := m.logger.With().Str("source", "metadataSyncLoop").Logger()
+	ctx := context.TODO()
+
+	ticker := time.NewTicker(metadataSyncPeriod)
+	for {
+		select {
+		case <-ticker.C:
+			logger.Info().Msg("Background metadata sync check triggered")
+			if err := m.broadcastMetadataRequests(); err != nil {
+				logger.Error().Err(err).Msg("Failed to send metadata requests")
+				return err
+			}
+
+		case <-ctx.Done():
```

Comment:
Where is this triggered from?

Reply:
Good call. Nothing stops this at the moment. Leaving this open for now and will add a comment once I'm back in the code.